arangod/RocksDBEngine/RocksDBOptimizerRules.cpp

```diff
@@ -218,7 +221,18 @@ void RocksDBOptimizerRules::reduceExtractionToProjectionRule(
           if (picked == nullptr || 
               picked->fields().size() > idx->fields().size()) {
             // found an index that would cover the projection
-            picked = idx;
+            if (hint.type() == aql::IndexHint::HintType::Simple) {
+              // index hint type "simple" allows us to use only one of the designated indexes
+              for (std::string const& hinted : hint.hint()) {
+                if (idx->name() == hinted) {
+                  picked = idx;
+                  break;
```

Comment:
I'm not sure this will honor the hint prioritization. Doesn't it depend on the ordering imposed by the outer loop instead? I think what we need to do is first iterate over the hinted indices in order and see if one can cover the projection. If not, then iterate over the rest of the list.